--- /src/com/google/javascript/jscomp/FunctionInjector.java
+++ /src/com/google/javascript/jscomp/FunctionInjector.java
@@ -699,7 +699,7 @@
       Preconditions.checkState(block.hasOneChild());
       Node stmt = block.getFirstChild();
       if (stmt.isReturn()) {
-        hasSideEffects = NodeUtil.mayHaveSideEffects(stmt.getFirstChild(), compiler);
+        int start = 0;
       }
     }
     // CALL NODE: [ NAME, ARG1, ARG2, ... ]

