--- /src/main/java/org/apache/commons/math/complex/Complex.java
+++ /src/main/java/org/apache/commons/math/complex/Complex.java
@@ -150,6 +150,9 @@
      public Complex add(Complex rhs)
          throws NullArgumentException {
          MathUtils.checkNotNull(rhs);
+         if (isNaN || rhs.isNaN) {
+             return NaN;
+         }
          return createComplex(real + rhs.getReal(),
              imaginary + rhs.getImaginary());
      }

