--- /src/main/java/org/apache/commons/math3/genetics/ElitisticListPopulation.java
+++ /src/main/java/org/apache/commons/math3/genetics/ElitisticListPopulation.java
@@ -48,7 +48,9 @@
                                    final int populationLimit,
                                    final double elitismRate) {
         super(chromosomes, populationLimit);
-        this.elitismRate = elitismRate;
+        if (elitismRate < 0 || elitismRate > 1) {
+            throw new OutOfRangeException(LocalizedFormats.ELITISM_RATE, elitismRate, 0, 1);
+        }
     }
 
     /**
@@ -62,6 +64,9 @@
      */
     public ElitisticListPopulation(final int populationLimit, final double elitismRate) {
         super(populationLimit);
+	if (elitismRate < 0 || elitismRate > 1) {
+            throw new OutOfRangeException(LocalizedFormats.ELITISM_RATE, elitismRate, 0, 1);
+        }
         this.elitismRate = elitismRate;
     }
 
