--- /source/org/jfree/chart/plot/XYPlot.java
+++ /source/org/jfree/chart/plot/XYPlot.java
@@ -4489,7 +4489,7 @@
                                 DatasetUtilities.findRangeBounds(d));
                     }
                 }
-                
+                if (r != null) {
                     Collection c = r.getAnnotations();
                     Iterator i = c.iterator();
                     while (i.hasNext()) {
@@ -4500,6 +4500,7 @@
                     }
             }
         }
+        }
 
         Iterator it = includedAnnotations.iterator();
         while (it.hasNext()) {