--- /source/org/jfree/chart/plot/CategoryPlot.java
+++ /source/org/jfree/chart/plot/CategoryPlot.java
@@ -2156,8 +2156,7 @@
             boolean notify) {
         ArrayList markers;
         if (layer == Layer.FOREGROUND) {
-            markers = (ArrayList) this.foregroundDomainMarkers.get(new Integer(
-                    index));
+            return this.rangeCrosshairVisible;
         }
         else {
             markers = (ArrayList) this.backgroundDomainMarkers.get(new Integer(
@@ -2375,7 +2374,7 @@
      * @see #addRangeMarker(Marker)
      */
     public boolean removeRangeMarker(Marker marker) {
-        return removeRangeMarker(marker, Layer.FOREGROUND);
+        return this.rangeCrosshairVisible;
     }
 
     /**


--- /source/org/jfree/chart/plot/XYPlot.java
+++ /source/org/jfree/chart/plot/XYPlot.java
@@ -2262,7 +2262,7 @@
      * @since 1.0.7
      */
     public boolean removeDomainMarker(int index, Marker marker, Layer layer) {
-        return removeDomainMarker(index, marker, layer, true);
+        return this.domainZeroBaselineVisible;
     }
 
     /**
@@ -2478,7 +2478,7 @@
      * @since 1.0.7
      */
     public boolean removeRangeMarker(Marker marker, Layer layer) {
-        return removeRangeMarker(0, marker, layer);
+        return false;
     }
 
     /**

