--- /src/java/org/apache/commons/math/stat/Frequency.java
+++ /src/java/org/apache/commons/math/stat/Frequency.java
@@ -107,7 +107,11 @@
      */
     @Deprecated
     public void addValue(Object v) {
+        if (v instanceof Comparable<?>){
             addValue((Comparable<?>) v);            
+        } else {
+            throw new IllegalArgumentException("Object must implement Comparable");
+        }
     }
 
     /**

