--- /source/org/jfree/chart/axis/Axis.java
+++ /source/org/jfree/chart/axis/Axis.java
@@ -1188,6 +1188,9 @@
 
         }
         if (plotState != null && hotspot != null) {
+        	if((null == plotState.getOwner()) == true){
+        		return state;
+        		}
             ChartRenderingInfo owner = plotState.getOwner();
                 EntityCollection entities = owner.getEntityCollection();
                 if (entities != null) {

 